datafusion/functions/benches/to_char.rs

```diff
@@ -42,8 +38,8 @@ fn random_date_in_range(
     start_date + TimeDelta::try_days(random_days).unwrap()
 }
 
-fn data(rng: &mut ThreadRng) -> Date32Array {
-    let mut data: Vec<i32> = vec![];
+fn data(rng: &mut ThreadRng) -> Vec<Expr> {
```

Comment:
doesn't this change the meaning of the benchmark from testing to_char on a single array to testing to_char on a single row with 1000's of arguments?

Like `to_date(v1, v2, v3, .....)`

instead of `to_date(arr)` ?

Is that intentional?

Reply:
The benchmark functions iterate on the data and patterns so that shouldn't be the case. 
```
        let data = data(&mut rng);
        let patterns = patterns(&mut rng);

        b.iter(|| {
            data.iter().enumerate().for_each(|(idx, i)| {
                black_box(to_char(i.clone(), patterns.get(idx).unwrap().clone()));
            })
        })
```

Reply:
Right, but then the benchmark is simply testing how fast the `to_char` expr_fn function is creating a `Expr::Function` call (but not actually evaluting the expression)

```rust
data.iter().enumerate().for_each(|(idx, i)| {
   black_box(to_char(i.clone(), patterns.get(idx).unwrap().clone())); 
})
```

Reply:
The numbers looked amazing though :)